shell.nix

```
@@ -73,7 +73,7 @@ let
     ];
     extensions = [
       "rust-src" # required to compile the core library
-      "llvm-tools-preview" # currently required to support recently added flags
+      "llvm-tools" # currently required to support recently added flags
```

```suggestion
      "llvm-tools-preview" # currently required to support recently added flags
```

This breaks on Nix unfortunately. I tried to look into the root cause of this, and it seems that the Rust overlay we use extracts the extensions from the actual upstream Rust package, so it seems there's at least one point in the chain where this rename hasn't propagated.

I tested these changes with `llvm-tools-preview` and managed to compile a board / run a few makefile commands. Happy to see this merged without this particular change, and I can investigate it when I'm not on airport wifi.

---

I reverted the change to shell.nix.